src/Business Foundation/App/AuditCodes/src/SourceCodeSetup.Page.al

```diff
@@ -0,0 +1,51 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Foundation.AuditCodes;
+
+/// <summary>
+/// Source codes are used to categorize the source of a transaction.
+/// </summary>
+/// <remarks>
+/// This page is used to set up source codes that are used to categorize the source of a transaction. 
+/// Each feature that introduces a new transaction source should add a field to set up a default source code.
+/// </remarks>
+page 279 "Source Code Setup"
+{
+    ApplicationArea = All;
+    Caption = 'Source Code Setup';
+    DeleteAllowed = false;
+    InsertAllowed = false;
+    PageType = Card;
+    SourceTable = "Source Code Setup";
+    UsageCategory = Administration;
+
+    layout
+    {
+        area(factboxes)
+        {
+            systempart(Control1900383207; Links)
+            {
+                ApplicationArea = RecordLinks;
+                Visible = false;
+            }
+            systempart(Control1905767507; Notes)
+            {
+                ApplicationArea = Notes;
+                Visible = false;
+            }
+        }
+    }
+
+    trigger OnOpenPage()
```

Comment:
Is it a good idea to move this logic into the table.

And call it during init company as well?

Reply:
there is already a call in CompanyInitialize which sets up the setup record for most areas. The pattern of creating an empty setup record OnOpenPage if none exists is used in many places. I don't see how maving it to a table would help. I'd move the implementation into an impl. codeunit but it seems overkill in this case.